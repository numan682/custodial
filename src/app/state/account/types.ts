import { Transaction } from 'app/state/transaction/types'
import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export interface BalanceDetails {
  available: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  debonding: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  delegations: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  total: StringifiedBigInt | null
}

export interface Allowance {
  address: string
  amount: StringifiedBigInt
}

export interface Account extends BalanceDetails {
  address: string
  allowances?: Allowance[]
}

/* --- STATE --- */
export interface AccountState extends Account {
  loading: boolean
  accountError?: ErrorPayload
  transactions: Transaction[]
  transactionsError?: ErrorPayload
}
