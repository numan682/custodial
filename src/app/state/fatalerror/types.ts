/* --- STATE --- */
export interface FatalErrorPayload {
  message: string
  stack?: string
  sagaStack?: string
}

export interface FatalErrorState {
  error?: FatalErrorPayload
}
