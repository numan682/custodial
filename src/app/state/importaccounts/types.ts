import { ErrorPayload } from 'types/errors'
import { WalletType } from '../wallet/types'
import { BalanceDetails } from '../account/types'

/* --- STATE --- */
export interface ImportAccountsListAccount {
  address: string
  balance?: BalanceDetails
  path: number[]
  pathDisplay: string
  privateKey?: string
  publicKey: string
  selected: boolean
  type: WalletType
}

export enum ImportAccountsStep {
  Idle = 'idle',
  OpeningUSB = 'opening_usb',
  LoadingAccounts = 'loading_accounts',
  LoadingBalances = 'loading_balances',
}

export interface ImportAccountsState {
  accounts: ImportAccountsListAccount[]
  error?: ErrorPayload
  showAccountsSelectionModal: boolean
  accountsSelectionPageNumber: number
  step: ImportAccountsStep
}
