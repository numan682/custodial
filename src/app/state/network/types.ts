/* --- STATE --- */

export type NetworkType = 'local' | 'testnet' | 'mainnet'

export interface NetworkState {
  selectedNetwork: NetworkType

  /** Ticker symbol (ROSE, TEST, etc.) */
  ticker: string

  /** chainContext / Genesis Hash */
  chainContext: string

  /** Current epoch */
  epoch: number

  /** Minimum staking amount */
  minimumStakingAmount: number
}
