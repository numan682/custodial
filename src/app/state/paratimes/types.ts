import { StringifiedBigInt } from 'types/StringifiedBigInt'
import { ErrorPayload } from 'types/errors'
import { ParaTime } from '../../../config'

export enum TransactionFormSteps {
  TransferType = 'transferType',
  ParaTimeSelection = 'paraTimeSelection',
  TransactionRecipient = 'transactionRecipient',
  TransactionAmount = 'transactionAmount',
  TransactionConfirmation = 'transactionConfirmation',
  TransactionSummary = 'transactionSummary',
  TransactionError = 'transactionError',
}

export enum TransactionTypes {
  Deposit = 'deposit',
  Withdraw = 'withdraw',
}

export interface TransactionForm {
  amount: string
  confirmTransfer: boolean
  confirmTransferToValidator: boolean
  confirmTransferToForeignAccount: boolean
  defaultFeeAmount: string
  ethPrivateKey: string
  feeAmount: string
  feeGas: string
  paraTime?: ParaTime
  recipient: string
  type: TransactionTypes | undefined
}

export interface ParaTimesState {
  balance: StringifiedBigInt
  isLoading: boolean
  transactionError?: ErrorPayload
  transactionForm: TransactionForm
  transactionFormStep: TransactionFormSteps
}

export type Runtime = {
  address: string
  id: string
  decimals: number
  feeGas: bigint
}

export type ParaTimeTransaction = Pick<
  TransactionForm,
  'amount' | 'ethPrivateKey' | 'feeAmount' | 'feeGas' | 'recipient' | 'type'
>
