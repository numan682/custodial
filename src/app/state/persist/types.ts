import { RootState } from 'types'
// eslint-disable-next-line @typescript-eslint/ban-types
interface PreserveAliasName extends String {}

export interface KeyWithSalt {
  key: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type StringifiedKeyWithSalt = string & PreserveAliasName

export interface EncryptedObject {
  secretbox: Uint8Array
  nonce: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type EncryptedString = string & PreserveAliasName
// Hover to check if inferred variable type is EncryptedString (not string)
export const testPreserveAliasName = '{}' as EncryptedString

export interface SetUnlockedRootStatePayload {
  persistedRootState: PersistedRootState
  stringifiedEncryptionKey: StringifiedKeyWithSalt
}

/* --- STATE --- */
export interface PersistState {
  hasPersistedProfiles: boolean
  isPersistenceUnsupported: boolean
  loading: boolean
  stringifiedEncryptionKey: undefined | 'skipped' | StringifiedKeyWithSalt
  enteredWrongPassword: boolean
}

export interface PersistedRootState extends Pick<RootState, 'theme' | 'wallet' | 'network'> {}
export interface SyncedRootState extends Pick<RootState, 'theme' | 'wallet' | 'network' | 'persist'> {}
