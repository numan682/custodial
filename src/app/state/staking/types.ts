/* --- STATE --- */

import { NetworkType } from 'app/state/network/types'
import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export interface Validator {
  address: string
  name?: string
  nodeAddress: string
  escrow: StringifiedBigInt
  status: 'active' | 'inactive' | 'unknown'
  rank: number
  media?: ValidatorMediaInfo
  current_rate?: number
  current_commission_bound?: CommissionBound
}

export interface Validators {
  timestamp: number
  network: NetworkType
  list: Validator[]
}

export interface CommissionBound {
  lower: number
  upper: number
  epochStart: number
  epochEnd?: number
}

export interface ValidatorDetails {
  currentCommissionBounds?: CommissionBound
  scheduledCommissionBounds?: CommissionBound[]
  nextCommission?: {
    fee: number
    epochStart: number
  }
}
export interface ValidatorMediaInfo {
  website_link?: string
  email_address?: string
  twitter_acc?: string
  tg_chat?: string
  logotype?: string
}

export interface Delegation {
  validator?: Validator
  amount: StringifiedBigInt
  shares: StringifiedBigInt
  validatorAddress: string
}

export interface DebondingDelegation extends Delegation {
  epoch: number
}

export interface StakingState {
  /** List of all the validators */
  validators: Validators | null

  /** Error from last attempt to update our list of validators */
  updateValidatorsError?: ErrorPayload

  /** List of active delegations for the selected account */
  delegations: Delegation[] | null

  /** List of debonding delegations for the selected account */
  debondingDelegations: DebondingDelegation[] | null

  /** Error from fetching delegations */
  updateDelegationsError?: ErrorPayload

  /** Address of the selected validator */
  selectedValidator?: string | null

  /** Lazy-loaded validator details (commission bounds and rates schedules) */
  selectedValidatorDetails: ValidatorDetails | null

  loading: boolean
}
