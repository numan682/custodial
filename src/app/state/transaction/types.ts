import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export enum TransactionType {
  StakingTransfer = 'staking.Transfer',
  StakingAddEscrow = 'staking.AddEscrow',
  StakingReclaimEscrow = 'staking.ReclaimEscrow',
  StakingAmendCommissionSchedule = 'staking.AmendCommissionSchedule',
  StakingAllow = 'staking.Allow',
  StakingWithdraw = 'staking.Withdraw',
  RoothashExecutorCommit = 'roothash.ExecutorCommit',
  RoothashExecutorProposerTimeout = 'roothash.ExecutorProposerTimeout',
  RegistryDeregisterEntity = 'registry.DeregisterEntity',
  RegistryRegisterEntity = 'registry.RegisterEntity',
  RegistryRegisterNode = 'registry.RegisterNode',
  RegistryRegisterRuntime = 'registry.RegisterRuntime',
  GovernanceCastVote = 'governance.CastVote',
  GovernanceSubmitProposal = 'governance.SubmitProposal',
  BeaconPvssCommit = 'beacon.PVSSCommit',
  BeaconPvssReveal = 'beacon.PVSSReveal',
  BeaconVrfProve = 'beacon.VRFProve',
  ConsensusDeposit = 'consensus.Deposit',
  ConsensusWithdraw = 'consensus.Withdraw',
  ConsensusAccountsParameters = 'consensus.Parameters',
  ConsensusBalance = 'consensus.Balance',
  ConsensusAccount = 'consensus.Account',
}

export interface Transaction {
  amount: StringifiedBigInt | undefined
  fee: StringifiedBigInt | undefined
  from: string | undefined
  hash: string
  level: number | undefined
  status: boolean | undefined
  timestamp: number | undefined
  to: string | undefined
  type: TransactionType
  // These are undefined on consensus transaction
  // Only appear on ParaTime transaction
  runtimeName: string | undefined
  runtimeId: string | undefined
  round: number | undefined
}

/* --- STATE --- */
export interface TransactionState {
  step?: TransactionStep
  active: boolean
  success: boolean
  error?: ErrorPayload
  preview?: TransactionPreview
}

export type NewTransactionType = 'transfer' | 'addEscrow' | 'reclaimEscrow'
export type TransactionPayload = TransferPayload | AddEscrowPayload | ReclaimEscrowPayload

export interface TransactionPreview {
  transaction: TransactionPayload
  fee?: StringifiedBigInt
  gas?: StringifiedBigInt
}

/**
 * Flow for a transaction to be sent.
 */
export enum TransactionStep {
  Building = 'building',
  Preview = 'preview',
  Signing = 'signing',
  Submitting = 'submitting',
  Sent = 'sent',
}

export interface TransferPayload {
  type: 'transfer'

  /* bech32 Address */
  to: string

  /* Token amount */
  amount: StringifiedBigInt
}

export interface AddEscrowPayload {
  type: 'addEscrow'

  /* bech32 Address */
  validator: string

  /* Token amount */
  amount: StringifiedBigInt
}

export interface ReclaimEscrowPayload {
  type: 'reclaimEscrow'

  /* bech32 Address */
  validator: string

  /* Shares to be reclaimed */
  shares: StringifiedBigInt

  /* Displayed token equivalent */
  amount: StringifiedBigInt
}
