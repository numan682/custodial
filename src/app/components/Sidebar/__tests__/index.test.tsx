import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import * as React from 'react'
import { Provider } from 'react-redux'
import { configureAppStore } from 'store/configureStore'
import { Wallet } from 'app/state/wallet/types'
import {
  useParaTimesNavigation,
  ParaTimesNavigationHook,
} from '../../../pages/ParaTimesPage/useParaTimesNavigation'
import { Navigation } from '..'

jest.mock('../../../pages/ParaTimesPage/useParaTimesNavigation')

const renderComponent = (store: any) => {
  return render(
    <Provider store={store}>
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>
    </Provider>,
  )
}

describe('<Navigation />', () => {
  let store: ReturnType<typeof configureAppStore>
  const mockUseParaTimesNavigationResult = {
    canAccessParaTimesRoute: false,
    getParaTimesRoutePath: (address: string) => address,
    paraTimesRouteLabel: 'MockParaTimesLabel',
  } as ParaTimesNavigationHook

  beforeEach(() => {
    jest.mocked(useParaTimesNavigation).mockReturnValue(mockUseParaTimesNavigationResult)
    store = configureAppStore({})
  })

  it('should match snapshot', () => {
    const component = renderComponent(store)
    expect(component.container.firstChild).toMatchSnapshot()
  })

  it('should render paraTime link', () => {
    jest.mocked(useParaTimesNavigation).mockReturnValue({
      ...mockUseParaTimesNavigationResult,
      canAccessParaTimesRoute: true,
    })
    renderComponent(
      configureAppStore({
        wallet: {
          selectedWallet: 'dummy',
          wallets: {
            dummy: {
              address: 'dummy',
            } as Wallet,
          },
        },
      }),
    )

    expect(screen.getByText('MockParaTimesLabel')).toBeInTheDocument()
    expect(screen.getByLabelText('Inherit')).toBeInTheDocument()
  })
})
