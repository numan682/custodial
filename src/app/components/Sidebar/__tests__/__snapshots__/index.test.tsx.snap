// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should match snapshot 1`] = `
.c11 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c11 g {
  fill: inherit;
  stroke: inherit;
}

.c11 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c11 *[stroke*='#'],.c11 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c11 *[fill-rule],
.c11 *[FILL-RULE],
.c11 *[fill*='#'],.c11 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c4 {
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  object-fit: contain;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background: component-sidebar;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100%;
  width: 220px;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 0px;
  padding-right: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 48px;
  width: 48px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 100%;
  overflow: hidden;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 0px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background: background-oasis-blue;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 24px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 24px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  padding: 12px;
}

.c5 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c7 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  height: 12px;
}

.c12 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 24px;
}

.c6 {
  font-size: 18px;
  line-height: 24px;
}

.c14 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  width: 100%;
}

.c14:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c14:focus > circle,
.c14:focus > ellipse,
.c14:focus > line,
.c14:focus > path,
.c14:focus > polygon,
.c14:focus > polyline,
.c14:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c14:focus::-moz-focus-inner {
  border: 0;
}

.c14:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c14:focus:not(:focus-visible) > circle,.c14:focus:not(:focus-visible) > ellipse,
.c14:focus:not(:focus-visible) > line,.c14:focus:not(:focus-visible) > path,
.c14:focus:not(:focus-visible) > polygon,.c14:focus:not(:focus-visible) > polyline,
.c14:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c14:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c17 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c17:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c17:focus > circle,
.c17:focus > ellipse,
.c17:focus > line,
.c17:focus > path,
.c17:focus > polygon,
.c17:focus > polyline,
.c17:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c17:focus::-moz-focus-inner {
  border: 0;
}

.c17:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c17:focus:not(:focus-visible) > circle,.c17:focus:not(:focus-visible) > ellipse,
.c17:focus:not(:focus-visible) > line,.c17:focus:not(:focus-visible) > path,
.c17:focus:not(:focus-visible) > polygon,.c17:focus:not(:focus-visible) > polyline,
.c17:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c17:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-top: 12px;
    padding-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-left: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-right: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c19 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c20 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    width: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    height: 6px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      href="/"
    >
      <div
        class="c2"
      >
        <div
          class="c3 "
          role="figure"
        >
          <img
            class="c4"
            role="presentation"
            src="IMAGE_MOCK"
          />
        </div>
        <div
          class="c5"
        />
        <span
          class="c6"
        >
          Oasis Wallet
        </span>
      </div>
    </a>
  </div>
  <div
    class="c7"
  />
  <div
    class="c8"
  >
    <nav
      class="c9"
    >
      <a
        aria-current="page"
        aria-label="menu.home"
        class="active"
        data-testid="nav-home"
        href="/"
      >
        <div
          class="c10"
        >
          <svg
            aria-label="Home"
            class="c11"
            viewBox="0 0 24 24"
          >
            <path
              d="m1 11 11-9 11 9m-8 12v-8H9v8m-5 0V9m16 14V9"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
          <div
            class="c12"
          />
          <span
            class="c6"
          >
            menu.home
          </span>
        </div>
      </a>
      <div
        class="c7"
      />
      <div
        class="c7"
      />
      <div
        class="c7"
      />
    </nav>
  </div>
  <div
    class="c7"
  />
  <nav
    class="c13"
  >
    <button
      aria-label="theme.darkMode"
      class="c14"
      type="button"
    >
      <div
        class="c15"
      >
        <svg
          aria-label="Moon"
          class="c11"
          viewBox="0 0 24 24"
        >
          <path
            d="M9.874 5.008c2.728-1.68 6.604-1.014 8.25.197-2.955.84-5.11 3.267-5.242 6.415-.18 4.28 3.006 6.588 5.24 7.152-1.964 1.343-4.36 1.293-5.235 1.172-3.568-.492-6.902-3.433-7.007-7.711-.106-4.278 2.573-6.35 3.994-7.225z"
            fill="none"
            stroke="#000"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <div
          class="c12"
        />
        <span
          class="c6"
        >
          theme.darkMode
        </span>
      </div>
    </button>
    <div
      class="c7"
    />
    <div
      class="c16"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="Open Menu"
        class="c17"
        type="button"
      >
        <div
          class="c18"
        >
          <div
            class="c19"
          >
            <svg
              aria-label="Language"
              class="c11"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.87 15.07l-2.54-2.51.03-.03c1.74-1.94 2.98-4.17 3.71-6.53H17V4h-7V2H8v2H1v1.99h11.17C11.5 7.92 10.44 9.75 9 11.35 8.07 10.32 7.3 9.19 6.69 8h-2c.73 1.63 1.73 3.17 2.98 4.56l-5.09 5.02L4 19l5-5 3.11 3.11.76-2.04zM18.5 10h-2L12 22h2l1.12-3h4.75L21 22h2l-4.5-12zm-2.62 7l1.62-4.33L19.12 17h-3.24z"
                fill="#333"
              />
            </svg>
          </div>
          <div
            class="c20"
          >
            <span
              class="c6"
            >
              Language
            </span>
          </div>
          <div
            class="c19"
          >
            <svg
              aria-label="FormDown"
              class="c11"
              viewBox="0 0 24 24"
            >
              <path
                d="m18 9-6 6-6-6"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
      </button>
    </div>
    <div
      class="c7"
    />
    <a
      aria-label="GitHub"
      href="https://github.com/oasisprotocol/oasis-wallet-web"
      rel="noopener"
      target="_blank"
    >
      <div
        class="c15"
      >
        <svg
          aria-label="Github"
          class="c11"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.999 1C5.926 1 1 5.925 1 12c0 4.86 3.152 8.983 7.523 10.437.55.102.75-.238.75-.53 0-.26-.009-.952-.014-1.87-3.06.664-3.706-1.475-3.706-1.475-.5-1.27-1.221-1.61-1.221-1.61-.999-.681.075-.668.075-.668 1.105.078 1.685 1.134 1.685 1.134.981 1.68 2.575 1.195 3.202.914.1-.71.384-1.195.698-1.47-2.442-.278-5.01-1.222-5.01-5.437 0-1.2.428-2.183 1.132-2.952-.114-.278-.491-1.397.108-2.91 0 0 .923-.297 3.025 1.127A10.536 10.536 0 0 1 12 6.32a10.49 10.49 0 0 1 2.754.37c2.1-1.424 3.022-1.128 3.022-1.128.6 1.514.223 2.633.11 2.911.705.769 1.13 1.751 1.13 2.952 0 4.226-2.572 5.156-5.022 5.428.395.34.747 1.01.747 2.037 0 1.47-.014 2.657-.014 3.017 0 .295.199.637.756.53C19.851 20.979 23 16.859 23 12c0-6.075-4.926-11-11.001-11"
            fill="#333"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="c12"
        />
        <span
          class="c6"
        >
          GitHub
        </span>
      </div>
    </a>
  </nav>
</div>
`;
